pkg/login/social/azuread_oauth.go

```diff
@@ -92,6 +93,10 @@ func (s *SocialAzureAD) UserInfo(client *http.Client, token *oauth2.Token) (*Bas
 		isGrafanaAdmin = &grafanaAdmin
 	}
 
+	// setting the role to empty to reflect that we are not syncronizing with the external provider
+	if s.skipOrgRoleSync {
```

Comment:
the extraction should be avoided altogether instead of overriding the result (otherwise can result in errors with invalid role and block authentication)

As well IsGrafanaAdmin should be nil if skip sync is set

Reply:
we extract the groups and then return early instead: https://github.com/grafana/grafana/pull/60322/commits/91482bc1db4d3cd44e6d03595e5902faebcf473d#diff-1932ec0f0e9483143b8e2d48df6f864540ae8a15a938d40011e5ca2712391077R81

Does this look better? 🤔 

This allows for only role, grafanaAdmin assignment to be done later in the function call.

WDYT?